src/lib.rs

```diff
@@ -66,222 +36,460 @@ pub fn invoke(params: u32) -> u32 {
+    {
+        match FromPrimitive::from_u64(method) {
+            Some(Method::Constructor) => {
+                Self::constructor(rt, cbor::deserialize_params(params)?)?;
+                Ok(RawBytes::default())
+            }
+            Some(Method::Join) => {
+                let res = Self::join(rt, cbor::deserialize_params(params)?)?;
+                Ok(RawBytes::serialize(res)?)
+            }
+            None => Err(actor_error!(unhandled_message; "Invalid method")),
         }
+    }
+}
+
+#[cfg(test)]
+mod test {
+    use crate::{Actor, ConsensusType, ConstructParams, JoinParams, Method, State, Status};
+    use cid::Cid;
+    use fil_actors_runtime::runtime::Runtime;
+    use fil_actors_runtime::test_utils::MockRuntime;
+    use fil_actors_runtime::{actor_error, cbor, INIT_ACTOR_ADDR};
+    use fvm_ipld_encoding::RawBytes;
+    use fvm_shared::address::Address;
+    use fvm_shared::econ::TokenAmount;
+    use fvm_shared::error::ExitCode;
+    use ipc_gateway::MIN_COLLATERAL_AMOUNT;
 
-        // release from SCA
-        if st.status != Status::Terminating {
-            st.send(
-                &Address::new_id(ext::sca::SCA_ACTOR_ADDR),
-                Method::ReleaseStake as u64,
-                RawBytes::serialize(FundParams {
-                    value: stake.clone(),
-                })?,
-                TokenAmount::zero(),
-            )?;
+    // just a test address
+    const IPC_GATEWAY_ADDR: u64 = 1024;
+
+    fn std_construct_param() -> ConstructParams {
+        ConstructParams {
+            parent: Default::default(),
+            name: "".to_string(),
+            ipc_gateway_addr: IPC_GATEWAY_ADDR,
+            consensus: ConsensusType::Delegated,
+            min_validator_stake: Default::default(),
+            min_validators: 0,
+            finality_threshold: 0,
+            check_period: 0,
+            genesis: vec![],
         }
+    }
 
-        // remove stake from balance table
-        st.rm_stake(&caller, &stake)?;
+    fn construct_runtime() -> MockRuntime {
+        let caller = *INIT_ACTOR_ADDR;
+        let receiver = Address::new_id(1);
+        let mut runtime = MockRuntime::new(caller, receiver);
 
-        // send back to owner
-        st.send(&caller, METHOD_SEND, RawBytes::default(), stake)?;
+        let params = std_construct_param();
 
-        st.mutate_state();
-        st.save();
-        Ok(None)
+        runtime.expect_validate_caller_addr(vec![caller]);
+
+        runtime
+            .call::<Actor>(
+                Method::Constructor as u64,
+                &cbor::serialize(&params, "test").unwrap(),
+            )
+            .unwrap();
+
+        runtime
     }
 
-    fn kill() -> anyhow::Result<Option<RawBytes>> {
-        let mut st = State::load();
+    #[test]
+    fn test_constructor() {
+        let runtime = construct_runtime();
+        assert_eq!(runtime.state.is_some(), true);
+        // let state: State = runtime.state().unwrap();
+    }
```

Comment:
Ideally we should make a `state` check here to double-check that the state was initialized correctly in the actor.